src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -168,9 +171,12 @@ public function hasItem($key)
         if (isset($this->deferred[$key])) {
             $item = (array) $this->deferred[$key];
             try {
-                $e = null;
+                $e = $ok = null;
                 $value = $item[CacheItem::CAST_PREFIX.'value'];
-                $ok = $this->doSave(array($key => $value), $item[CacheItem::CAST_PREFIX.'lifetime']);
+                $lifetime = $item[CacheItem::CAST_PREFIX.'expiry'];
+                if (0 <= $lifetime = null !== $lifetime ? $lifetime - time() ?: -1 : 0) {
```

Comment:
I don't know if others agree, but I find this line very hard to read+understand:

```php
if (0 <= $lifetime = null !== $lifetime ? $lifetime - time() ?: -1 : 0)
```